### PR TITLE
Custom Log wildcards issue fix

### PR DESCRIPTION
**Pull request by shpimpal**
This will fix the issue with custom log solution where files specified with wildcard pattern do not get picked up until agent restart and sometimes do not picked up at all as files are not readable to omsagent user.

- Moving the code that lists files with given pattern to "tailfilereader.rb" as we need sudo to list them

### CODE REVIEW

File: source/code/plugins/tailfilereader.rb [hunk 1]
```diff
@@ -365,7 +395,7 @@ def read_inode
     end
   end.parse!
 
-  a = Tailscript::NewTail.new(ARGV)
+  a = Tailscript::NewTail.new(ARGV[0])
```
Comment (simathih) [inline source/code/plugins/tailfilereader.rb]: are we sure all callers to this script pass an array? and not only sudo was passing Array?

File: source/code/plugins/tailfilereader.rb [hunk 2]
```diff
@@ -20,8 +20,38 @@ def initialize(paths)
...
+          expanded_paths << file unless file.nil?
+        end
+      }
+      return expanded_paths
+    end
+
     def start
-      start_watchers(@paths) unless @paths.empty?
+      paths = expand_paths()
+      start_watchers(paths) unless paths.empty?
```
Comment (simathih) [inline source/code/plugins/tailfilereader.rb]: earlier @paths passed to start_watchers was a space separated string, now it is space separated?

File: source/code/plugins/tailfilereader.rb [hunk 3]
```diff
@@ -20,8 +20,38 @@ def initialize(paths)
...
+          file = file_exists(path)
+          expanded_paths << file unless file.nil?
+        end
+      }
+      return expanded_paths
+    end
+
     def start
-      start_watchers(@paths) unless @paths.empty?
+      paths = expand_paths()
```
[the remark this answers is not in the document]
Comment (shpimpal) [reply source/code/plugins/tailfilereader.rb]: In ruby, instance variables start with @ and local variables are plain characters (or start with _). So it should not cause any conflicts.

File: source/code/plugins/tailfilereader.rb (hunk 1 above)
Comment (shpimpal) [reply source/code/plugins/tailfilereader.rb]: AFAIK, this plugin itself and its test plugin are the only ones that call this class.
What do you mean by this - "and not only sudo was passing Array?" ?

**Pull request merged**